pyomo/contrib/doe/doe.py

```diff
@@ -991,179 +776,140 @@ def _create_doe_model(self, no_obj=True):
         Parameters:
         -----------
         no_obj: if True, objective function is 0.
-        self.design_values: a dict of dictionaries, keys are the name of design variables, 
-        values are a dict where keys are the time points, values are the design variable value at that time point
-        self.optimize: if True, solve the problem unfixing the design variables. if False, solve the problem as a
-        square problem
-        self.objective_option: choose from 'det' or 'trace'. Optimization problem maximizes determinant or trace.
-        self.scale_nominal_param_value: if True, scale FIM but not scale Jacobian. This toggle can be opened for better performance when the
-        problem is poorly scaled.
-        self.tee_opt: if True, print IPOPT console output
-        self.Cholesky_option: if true, cholesky decomposition is used for Objective function (to optimize determinant). 
-            If true, determinant will not be calculated.
-            self.L_LB: if FIM is P.D., the diagonal element should be positive, so we can set a LB like 1E-10
-            self.L_initial: initialize the L
-        self.formula: choose from 'central', 'forward', 'backward', None
-        self.step: Sensitivity perturbation step size, a fraction between [0,1]. default is 0.001
 
         Return:
         -------
         m: the DOE model
         """
-        # call generator function to get scenario dictionary
-        scena_gen = Scenario_generator(self.param, formula=self.formula, step=self.step, store=True)
-        scenario_all = scena_gen.simultaneous_scenario()
         
-        # create model
-        m = self.create_model(scenario_all, args= self.args)
-        # discretize if discretization function is provided
+        # create scenario information for block scenarios
+        scena_object = ScenarioGenerator(self.param, formula=self.formula, step=self.step)
+        scena_gen = scena_object.generate_scenario()
+            
+        # a list of dictionary, each one is a parameter dictionary with perturbed parameter values
+        self.scenario_list = scena_gen["scenario"]
+        # dictionary, keys are parameter name, values are a list of scenario index where this parameter is perturbed.
+        self.scenario_num = scena_gen["scena_num"]
+        # dictionary, keys are parameter name, values are the perturbation step 
+        self.eps_abs = scena_gen["eps-abs"]
+
+        # Create a global model 
```

Comment:
This looks like duplicate code to earlier in the file. What is the difference?

Reply:
make a private method here